### ISSUE TITLE
Loading data cancelled

### ISSUE THREAD

**Issue by diededh** [compomics/peptide-shaker#292]
I have an issue with loading any data from searchGUI, there are more threads where people seemed to have a similar issue but I have not found a suggestion to fix this that works for me.
I hope you can help me. 

_> What steps will reproduce the problem?_
I cannot open searghGUI output from the Tutorial (1.4  Browsing Identification Results).

_> What is the expected output? What do you see instead?_
'Importing data cancelled'

_> What version of the product are you using? On what operating system?_
I am using PeptideShaker-1.16.15 on
a laptop running on windows 10, 8.00 GB RAM, Intel(R) Core(TM), i7-5500U CPU @2.40GHz

_> Please provide any additional information below._
**Report:**
Sun Nov 19 17:18:48 EST 2017        Import process for Tutorial (Sample: HeLa, Replicate: 0)

Sun Nov 19 17:18:48 EST 2017        Importing sequences from uniprot-human-reviewed-trypsin-june-2017_concatenated_target_decoy.fasta.
Sun Nov 19 17:19:17 EST 2017        FASTA file import completed.
Sun Nov 19 17:19:17 EST 2017        Importing gene mappings.
Sun Nov 19 17:19:19 EST 2017        Establishing local database connection.
Sun Nov 19 17:19:20 EST 2017        Reading identification files.
Sun Nov 19 17:19:20 EST 2017        Parsing qExactive01819.omx.
Sun Nov 19 17:19:25 EST 2017        Loading spectra for qExactive01819.omx.
Sun Nov 19 17:19:25 EST 2017        Importing qExactive01819.mgf
Sun Nov 19 17:19:25 EST 2017        qExactive01819.mgf imported.
Sun Nov 19 17:19:25 EST 2017        Collecting peptides to map.
Sun Nov 19 17:19:25 EST 2017        Mapping peptides to proteins.
Sun Nov 19 17:19:25 EST 2017        Importing PSMs from qExactive01819.omx
Sun Nov 19 17:19:26 EST 2017        PeptideShaker is encountering memory issues! See http://compomics.github.io/projects/peptide-shaker.html for help.
Sun Nov 19 17:19:26 EST 2017        Reducing Memory Consumption.
Sun Nov 19 17:19:29 EST 2017        60 identified spectra (1.0%) did not present a valid peptide.
Sun Nov 19 17:19:29 EST 2017        2544 of the best scoring peptides were excluded by the import filters:
Sun Nov 19 17:19:29 EST 2017            - 93.1% peptide length less than 8 or greater than 30.
Sun Nov 19 17:19:29 EST 2017            - 6.8% peptide presenting high mass or isotopic deviation.
Sun Nov 19 17:19:29 EST 2017        Parsing qExactive01819.t.xml.
Sun Nov 19 17:19:32 EST 2017        Loading spectra for qExactive01819.t.xml.
Sun Nov 19 17:19:32 EST 2017        Collecting peptides to map.
Sun Nov 19 17:19:32 EST 2017        Mapping peptides to proteins.
Sun Nov 19 17:19:32 EST 2017        Importing PSMs from qExactive01819.t.xml
Sun Nov 19 17:19:32 EST 2017        An error occurred: null
Sun Nov 19 17:19:32 EST 2017        Please contact the developers (https://github.com/compomics/peptide-shaker/issues).

Sun Nov 19 17:19:32 EST 2017        Importing Data Canceled!

**Bug report**:
Sun Nov 19 17:18:08 EST 2017: PeptideShaker version 1.16.15.
java.lang.NullPointerException
	at com.compomics.util.db.ObjectsCache.addObject(ObjectsCache.java:313)
	at com.compomics.util.db.ObjectsDB.retrieveObject(ObjectsDB.java:1049)
	at com.compomics.util.db.ObjectsDB.retrieveObject(ObjectsDB.java:927)
	at com.compomics.util.db.ObjectsDB.retrieveObject(ObjectsDB.java:889)
	at com.compomics.util.experiment.identification.IdentificationDB.getAssumptions(IdentificationDB.java:436)
	at com.compomics.util.experiment.identification.Identification.getAssumptions(Identification.java:982)
	at com.compomics.util.experiment.identification.Identification.getAssumptions(Identification.java:1002)
	at eu.isas.peptideshaker.fileimport.PsmImporter.importAssumptions(PsmImporter.java:688)
	at eu.isas.peptideshaker.fileimport.PsmImporter.importPsm(PsmImporter.java:384)
	at eu.isas.peptideshaker.fileimport.PsmImporter.access$000(PsmImporter.java:68)
	at eu.isas.peptideshaker.fileimport.PsmImporter$PsmImporterRunnable.run(PsmImporter.java:1466)
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.util.concurrent.FutureTask.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)

Thank you andvance!

**Comment by hbarsnes** [comment]
Hi,

How much memory are you providing to PeptideShaker? With 8 GBs available you should have more than enough to load the example data.

You can check and increase the amount of memory given to PeptideShaker via the Welcome Dialog > Settings & Help > Settings > Java Settings. Note that you need Java 64 bit to use more than 2 GB of memory. For more details see: http://compomics.github.io/projects/compomics-utilities/wiki/javatroubleshooting.html.

Best regards,
Harald

**Comment by diededh** [comment]
Hi Harald,

Thank you for getting back to me.
I am using the 64 bit Java and allow the software to use 6.9 GB of my memory:
![image](https://user-images.githubusercontent.com/18311324/33041434-48e2192a-ce0c-11e7-8915-6cfb7f537ba6.png)

So I am not sure that this is a memory issue?

Best,
Diede

**Comment by hbarsnes** [comment]
Hi Diede,

I agree that you should have more than enough memory for the example dataset. Which is why the "PeptideShaker is encountering memory issues!" message (followed by "Reducing Memory Consumption") in the report is so peculiar. Did you try rerunning the loading of the data in PeptideShaker after ensuring that you are giving it enough memory? Or did the memory setting not change?

The actual error in the bug report is due to an error in our cache handling, where the temporary information stored in the cache could get out of sync with the information stored in the underlying database. This has been corrected, but there will still be some time before we can release a new version as we have to update the rest of the code as well. I will let you know as soon as the new version becomes available for testing.

Best regards,
Harald

**Comment by hbarsnes** [comment]
This issue should hopefully be solved in the new beta releases of SearchGUI and PeptideShaker. Please see https://groups.google.com/forum/#!topic/peptide-shaker/1ecY0IyMOBM for more details.

If the issue still exists in the beta releases, please open a new issue at https://github.com/compomics/peptide-shaker-2.0-issue-tracker/issues.

If the issue has been solved, it would also be great of you could let us know by replying to this original issue and letting us know. :)

**Comment by hbarsnes** [comment]
This issue should hopefully be fixed in the new release that is available now. If not, please let us know by opening a new issue.
